ghcide/src/Development/IDE/Core/Actions.hs

```diff
@@ -33,18 +41,92 @@ import           Language.LSP.Protocol.Types          (DocumentHighlight (..),
                                                        SymbolInformation (..),
                                                        normalizedFilePathToUri,
                                                        uriToNormalizedFilePath)
+import           Language.LSP.Server                  (resRootPath)
+import           System.Directory                     (createDirectoryIfMissing,
+                                                       doesFileExist,
+                                                       getPermissions,
+                                                       setOwnerExecutable,
+                                                       setOwnerWritable,
+                                                       setPermissions)
+import           System.FilePath                      (takeDirectory, (<.>),
+                                                       (</>))
 
 
--- | Eventually this will lookup/generate URIs for files in dependencies, but not in the
--- project. Right now, this is just a stub.
+-- | Generates URIs for files in dependencies, but not in the
+-- project. Dependency files are produced from an HIE file and
+-- placed in the .hls/dependencies directory.
```

Comment:
Can you please elaborate a little about what you find awkward about generating a new directory? I don't think putting things in dist-newstyle would be ideal because it's used by another tool entirely (cabal). As it is, the code handles `.hls/dependency` being wiped out just fine, but would have trouble if just part of it were corrupted.

Reply:
It's just One More Thing. At the moment users of HLS need to worry about:
- `dist-newstyle`
- `~/.cache/ghcide`
- and now `.hls`

Plus, it's in the project directory, so everyone will have to add a new thing to their `.gitignores`. It's not disastrous, but it is a real cost.

Does it need to be in the project root?

Reply:
I think it needs to be in the project root to prevent vscode from spawning another HLS instance.